packages/aws-cdk-lib/aws-cloudwatch/lib/alarm.ts

```diff
@@ -267,7 +267,7 @@ export class Alarm extends AlarmBase {
     if (ec2ActionsRegexp.test(actionArn)) {
       // Check per-instance metric
       const metricConfig = this.metric.toMetricConfig();
-      if (metricConfig.metricStat?.dimensions?.length != 1 || metricConfig.metricStat?.dimensions![0].name != 'InstanceId') {
+      if (metricConfig.metricStat?.dimensions?.length != 1 || !metricConfig.metricStat?.dimensions?.some(dimension => dimension.name === 'InstanceId')) {
```

Comment:
I've just stumbled upon this PR and I'm wondering if there shouldn't be `&&` operator instead of `||`? We want to throw the exception when there ale multiple dimensions AND none of them is `InstanceId`. Is that right?